src/lib.rs

```diff
@@ -384,6 +384,14 @@ impl DeviceHandle {
 	}
 }
 
+fn to_cstring_safe(ptr: *const std::os::raw::c_char) -> CString {
```

Comment:
I'd name this `cstring_from_ptr` - I don't see a reason to add the `safe` suffix (like, any function should be safe or marked "unsafe" otherwise)